ceph/utils.py

```diff
@@ -165,6 +165,33 @@ def volume_cleanup(volume, osp_cred):
                     return 1
 
 
+# Added this function to cleanup stale volumes with status deleting, available, error, unknown, reserved
+def cleanup_ceph_vols(osp_cred, pattern=None, timeout=300):
+    driver = get_openstack_driver(osp_cred)
+    timeout = datetime.timedelta(seconds=timeout)
+    for volume in driver.list_volumes():
+        log.info("volume %s status is %s", volume.name, volume.state)
+    with parallel() as p:
+        for volume in driver.list_volumes():
+            if volume.state in "deleting":
```

Comment:
IMO, it would be nice if vol state list is available and any vol state which falls could be considered for deletion.
```
vol_states = [ "deleting", "unknown", . , .. ,]
if vol.state in vol_states:
   # consider it for deletion or any required action.
   p.spawn(<def>, args)
```

Reply:
Update the states to use tuple.